### PR TITLE
bioformats2raw 0.2.6, use PKG_NAME-PKG_VERSION-PKG_BUILDNUM

### PR DESCRIPTION
**Pull request by manics**
- Update bioformats2raw to 0.2.6
- GitHub workflow for testing
- Installs files to share/PKG_NAME-PKG_VERSION-PKG_BUILDNUM
- Restrict Windows to openjdk=8 https://github.com/ome/conda-bioformats2raw/issues/3
- Patch windows bat file to avoid CLASSPATH length being too long for cmd
- Splits the package into a large noarch `bioformats2raw-libs` package (140 MB), and several smaller arch-specific `bioformats2raw` packages containing wrapper scripts. A free Anaconda Cloud account only provides 3GB space: https://docs.anaconda.com/anaconda-cloud/faq/#what-is-included-in-the-free-version-of-cloud, and this avoids having three copies of the jars. In theory conda supports sub-packages, but I don't know if it's possible to have mixed noarch and *-64 subpackages.

To upload this requires the following GitHub secrets:
- `ANACONDA_USER`
- `ANACONDA_API_TOKEN`

Tag: `0.2.6-0`

### CODE REVIEW

File: bld.bat [hunk 1]
```diff
@@ -1,5 +1,9 @@
-mkdir %LIBRARY_PREFIX%\bin
-mkdir %LIBRARY_PREFIX%\lib
-copy %SRC_DIR%\bin\*.bat %LIBRARY_PREFIX%\bin
-copy %SRC_DIR%\lib\*.jar %LIBRARY_PREFIX%\lib
+set pkg=%PKG_NAME%-%PKG_VERSION%-%PKG_BUILDNUM%
+
+mkdir %PREFIX%\bin
```
Comment (manics) [inline bld.bat]: No idea whether the standard is `PREFIX` or `LIBRARY_PREFIX`, and/or whether it's the same or different on unix/windows.

File: windows-bat-classpath-wildcard.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
++set CLASSPATH=%APP_HOME%\lib\*
```
Comment (manics) [inline windows-bat-classpath-wildcard.patch]: Using the wildcard format for the classpath avoids a command-too-long error from Windows cmd if `APP_HOME` is too long.

**Comment by manics** [comment]
Example upload here:
- https://github.com/manics/conda-bioformats2raw/releases/tag/tag-test
- https://anaconda.org/manics/bioformats2raw-libs/files
- https://anaconda.org/manics/bioformats2raw/files

File: bld.bat (hunk 1 above)
Comment (joshmoore) [reply bld.bat]: Pretty sure I had tried `PREFIX` first. But perhaps enough has changed here.

**Review by joshmoore — APPROVED** [review]
LGTM. Depending on how we progress with conda-forge packages in general, it may make sense to open this work against the bioformats2raw repository directly to not need to bump the version number and checksum, but we can address that later. Definitely good to have the library caught up. Thanks!

@sbesson: anything you want to add in terms of recent template decisions?

**Review by sbesson — APPROVED** [review]
:+1: from my side. Might need other examples of conda build/upload to start driving a generic template workflow. Immediately LGTM and test a tag workflow

**Comment by manics** [comment]
For anyone tagging be aware that from past experience the token we've used for conda releases allows uploads to an existing repository but not the reation of a new one. This means `bioformats2raw-libs` will need to have a placeholder package uploaded first.

**Comment by joshmoore** [comment]
So the relisting of this means you think it's time for it to be tagged & released? Are you blocked from doing that? If so, should we try to get rid of those blockers? Otherwise, :+1:

**Comment by manics** [comment]
I'll need:
- [x] commit privileges on this repo
- access to the anaconda credentials or token for uploading a package (this will be added as a GitHub workflow secret)
- ability to create a new `ome` anaconda repository if we're sticking with the current workflow for our conda repos where the secret token that can upload conda packages to an existing repo on Anaconda cloud, but not create new ones under the `ome` org.

**Comment by manics** [comment]
Last commit adds a workflow badge to the README

**Comment by camlloyd** [comment]
Hello, will 0.2.6 be uploaded to Anaconda Cloud soon, or will this be skipped in favour of the upcoming 0.3.0? Thanks!

**Comment by jburel** [comment]
> For anyone tagging be aware that from past experience the token we've used for conda releases allows uploads to an existing repository but not

I have created a token this morning so we can push to the ome channel and the creation works see https://anaconda.org/ome/bftools-libs
i will vote to merge this PR and tag
Any objection?

**Comment by sbesson** [comment]
No objection from my side especially now we have successfully tested the deployment workflow with https://github.com/ome/conda-bftools/pull/6

**Comment by manics** [comment]
Probably worth noting that since https://github.com/ome/conda-bioformats2raw/commit/7919b9ea1bd831a527c40f21d20e542c155728af#diff-c01b381d8738b25f3fb172619b2a89fe3abc0950fcfcd26c7bab376abcc91087 is a patch file it'll fail to apply if an upstream update changes the list of jars, in other words you'd need to regenerate the patch. Ideally this would be dealt with in the upstream package.

**Comment by jburel** [comment]
I have pushed 2 commits since last viewed.
I will list tomorrow for review before tagging and deploying

**Review by joshmoore — APPROVED** [review]
LGTM

**Comment by jburel** [comment]
@camlloyd version 0.2.6 is now available on conda. https://anaconda.org/ome/bioformats2raw

**Comment by camlloyd** [comment]
@jburel Thank you! 🥳

**Pull request merged**